examples_tests/15.LayeredCubeMapShadows/deriv_map_gen.comp

```diff
@@ -1,97 +1,46 @@
 #version 450 core
 
-//place for #defines to put with sprintf
-%s
+layout (local_size_x = 16, local_size_y = 16) in;
 
-#if 0
-#extension GL_NV_compute_shader_derivatives : enable
-#if defined(GL_NV_compute_shader_derivatives)
-layout (derivative_group_quadsNV) in;
+layout (binding = 7) uniform sampler2D bumpMapSampler;
+layout (binding = 0, rg16_snorm) uniform image2D derivativeMapImage;
 
-#define IRR_DDX_IS_DEFINED 1
-#define IRR_DDY_IS_DEFINED 1
+layout (location = 0) uniform float uHeightScaleFactor = 0.001;
 
-float DDX(in float x)
-{
-	return dFdx(x);
-}
+shared float smem[324];//18*18
 
-float DDY(in float y)
+int getAddr(in ivec2 threadID)
```

Comment:
macro like `(18*(threadID.y+1) + threadID.x+1)` should work much better

Reply:
what's the difference?

Reply:
you've coded the function as-if there's real SIMD in a single GPU thread... so there are useless multiplies here also a cast to int for no reason.

The upside of a macro is that it can function like a template ;) (doesn't care if you use uint or int)

Reply:
well, i thought `dot` will make it more SIMD. The cast is because my driver is a little retarded and while compiling without cast, gives errors like `dot` is returning always float, same thing with `mix`.

>(doesn't care if you use uint or int)

well, that's actually source of i-dont-know-whats-going-on bugs :P

Reply:
if your compiler is complaining about dot being a float then you shouldn't use it... it might actually calc the index as a float XD

Reply:
it probably does, but they're integer's small enough to be representable on 32bit float

Reply:
but converting from float to int and back is not free